glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+class HttpPingUploader {
+    let LOG_TAG = "glean/HttpPingUploader"
+
+    func logPing(path: String, data: String, config: Configuration) {
+        if config.logPings {
+            NSLog("\(LOG_TAG) : \(path)\n\(data)")
+        }
+    }
+
+    func upload(path: String, data: String, config: Configuration, callback: @escaping (Bool, Error?) -> Void) {
+        logPing(path: path, data: data, config: config)
+
+        if let request = buildRequest(path: path, data: data, config: config) {
+            let task = URLSession.shared.dataTask(with: request) { (_, response, error) in
+                let httpResponse = response as? HTTPURLResponse
+                let statusCode = httpResponse?.statusCode ?? 0
+                switch statusCode {
+                    case 200..<300:
+                        // Known success errors (2xx):
+                        // 200 - OK. Request accepted into the pipeline.
+
+                        // We treat all success codes as successful upload even though we only expect 200.
+                        callback(true, nil)
+                    case 400..<500:
+                        // Known client (4xx) errors:
+                        // 404 - not found - POST/PUT to an unknown namespace
+                        // 405 - wrong request type (anything other than POST/PUT)
+                        // 411 - missing content-length header
+                        // 413 - request body too large (Note that if we have badly-behaved clients that
+                        //       retry on 4XX, we should send back 202 on body/path too long).
+                        // 414 - request path too long (See above)
+
+                        // Something our client did is not correct. It's unlikely that the client is going
+                        // to recover from this by re-trying again, so we just log and error and report a
+                        // successful upload to the service.
+                        callback(true, error)
+                    default:
+                        // Known other errors:
+                        // 500 - internal error
+
+                        // For all other errors we log a warning an try again at a later time.
+                        callback(false, error)
+                }
+            }
+            task.resume()
+        }
+    }
+
+    func buildRequest(path: String, data: String, config: Configuration) -> URLRequest? {
+        if let url = URL(string: config.serverEndpoint + path) {
+            var request = URLRequest(url: url)
+            request.addValue("application/json; charset=utf-8", forHTTPHeaderField: "Content-Type")
```

Comment:
`addValue(..., forHTTPHeaderField: ...)` is a weird API o_O

Reply:
Yeah, it's pretty consistent across Swift to have an 'unnamed' first parameter and named parameters after that.